ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -98,6 +99,23 @@ static int azure_file_read(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t, size_t, voi
   __attribute__((__unused__));
 static int azure_file_size(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t *) __attribute__((__unused__));
 
+// Construct a pathname from the directory and the object name.
+static std::string
+azure_path(const std::string &dir, const std::string &name)
+{
+    // Skip over "./" and variations (".//", ".///./././//") at the beginning of the name.
+    int i = 0;
+    while (name[i] == '.') {
+        if (name[1] != '/')
+            break;
+        i += 2;
+        while (name[i] == '/')
+            i++;
+    }
```

Comment:
I don't know about this - we need to be more careful since we don't want to filter out something like a leading "../", which would be a different directory. I posted a link in the slack channel for what I think would be a better approach.

Reply:
resolved by using `std::filesystem::canonical`